ethcore/sync/src/chain/handler.rs

```diff
@@ -562,17 +562,33 @@ impl SyncHandler {
 
 	/// Called by peer to report status
 	fn on_peer_status(sync: &mut ChainSync, io: &mut dyn SyncIo, peer_id: PeerId, r: &Rlp) -> Result<(), DownloaderImportError> {
+		let mut r = r.iter();
 		sync.handshaking_peers.remove(&peer_id);
-		let protocol_version: u8 = r.val_at(0)?;
+		let protocol_version: u8 = r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?;
 		let warp_protocol_version = io.protocol_version(&WARP_SYNC_PROTOCOL_ID, peer_id);
 		let warp_protocol = warp_protocol_version != 0;
 		let private_tx_protocol = warp_protocol_version >= PAR_PROTOCOL_VERSION_3.0;
+		let network_id = r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?;
+		let difficulty = Some(r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?);
+		let latest_hash = r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?;
+		let genesis = r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?;
+		let forkid_validation_error = {
+			if protocol_version >= ETH_PROTOCOL_VERSION_64.0 {
+				let fork_id = r.next().ok_or(rlp::DecoderError::RlpIsTooShort)?.as_val()?;
+				sync.fork_filter.is_compatible(io.chain(), fork_id).err().map(|e| (fork_id, e))
+			} else {
+				None
```

Comment:
we should support both eth 63 and 64 and drop support for eth 62, see https://github.com/ethereum/go-ethereum/pull/20140

Reply:
never mind, I see that this is because `protocolVersion` is reused for warp protocol versioning as well